actix-identity/src/identity.rs

```diff
@@ -126,9 +124,12 @@ impl Identity {
     ///     }
     /// }
     /// ```
-    pub fn id(&self) -> Result<String, anyhow::Error> {
+    pub fn id(&self) -> Result<String, IdentityError> {
         self.0.session.get(ID_KEY)?.ok_or_else(|| {
-            anyhow!("Bug: the identity information attached to the current session has disappeared")
+            IdentityError::MissingIdentityError(
+                "Bug: the identity information attached to the current session has disappeared"
```

Comment:
I don't think this should be treated in the same way of an error in `get_identity`.
When `id` is called, we have already successfully extracted the identity from the current session once. I'd suggest using a different error type here.

Reply:
Is it worth decoupling the get operation and the bug handling instance? It seems like our choices are either split the interface, allowing for two error return types, or have an enum describing both potential error outcomes.

(Also totally possible I'm misunderstanding something)